cloud/aws/bin/pgadmin.py

```diff
@@ -0,0 +1,191 @@
+import os
+import sys
+import time
+import re
+import urllib.error
+import urllib.request
+
+from enum import Enum
+from typing import Callable
+
+from cloud.shared.bin.lib import terraform
+from cloud.shared.bin.lib.config_loader import ConfigLoader
+from cloud.aws.templates.aws_oidc.bin import resources
+from cloud.aws.templates.aws_oidc.bin.aws_cli import AwsCli
+
+
+def run(config: ConfigLoader):
+    os.environ["TF_VAR_pgadmin_cidr_allowlist"] = get_cidr_list()
+    os.environ["TF_VAR_pgadmin"] = "true"
+    run_terraform(config)
+
+    pgurl = f"{config.get_base_url()}:4433"
+    print(
+        "\npgadmin terraform deployment finished. Waiting for pgadmin service (some request failures are expected):"
+    )
+    wait_for_pgadmin_response(pgurl)
+
+    print(f"\npgadmin service is available. URL: {pgurl}")
+    print_secrets(config)
+
+    input(
+        "\nWARNING: it is strongly recommended to clean up pgadmin resources once they are no-longer needed.\n"
+        f"Run 'bin/deploy --tag={os.environ['TF_VAR_image_tag']}' to manually clean up pgadmin resources.\n\n"
+        "Waiting to clean up pgadmin resources.  Press enter to trigger cleanup...\n"
+    )
+
+    os.unsetenv("TF_VAR_pgadmin_cidr_allowlist")
+    os.unsetenv("TF_VAR_pgadmin")
+    run_terraform(config)
+
+
+def get_cidr_list() -> str:
```

Comment:
suggestion (blocking): it's not immediately apparent which of these free-floating functions are intended to be public and which are not. Please prepend a single underscore to the function names that aren't intended to be called outside of this file.

Additionally, code organization here would benefit from another class that encapsulates most of these free-floating functions.

Reply:
Non of these functions except for run() are expected to be used outside of this file.  Added leading _